src/server/common/helpers/i18next.test.js

```diff
@@ -0,0 +1,135 @@
+import { load } from 'cheerio'
+import { afterEach, beforeEach, describe, expect, it } from 'vitest'
+import { statusCodes } from '~/src/server/common/constants/status-codes.js'
+import { createServer } from '~/src/server/index.js'
+
+describe('#i18nPlugin - integration', () => {
+  /** @type {Server} */
+  let server
+
+  beforeEach(async () => {
+    server = await createServer()
+    await server.initialize()
+  })
+
+  afterEach(async () => {
+    await server.stop({ timeout: 0 })
+  })
+
+  describe('language detection and html lang attribute', () => {
+    it('should default to english for root path', async () => {
+      const response = await server.inject({
+        method: 'GET',
+        url: '/'
+      })
+
+      expect(response.statusCode).toBe(statusCodes.ok)
+
+      expect(response.headers['content-type']).toContain('text/html')
+
+      const $ = load(response.result)
+
+      expect($('html').attr('lang')).toBe('en')
+    })
+
+    it('should set welsh language for /cy path', async () => {
+      const response = await server.inject({
+        method: 'GET',
+        url: '/cy'
+      })
+
+      expect(response.statusCode).toBe(statusCodes.ok)
+
+      const $ = load(response.result)
+
+      expect($('h1').first().text()).toBe('Sites')
+      expect($('html').attr('lang')).toBe('cy')
+    })
+
+    it.each([
+      { url: '/', expectedLang: 'en', description: 'english' },
+      { url: '/cy', expectedLang: 'cy', description: 'welsh' }
+    ])(
+      'should set lang="$expectedLang" for $description pages ($url)',
+      async ({ url, expectedLang }) => {
+        const response = await server.inject({
+          method: 'GET',
+          url
+        })
+
+        expect(response.statusCode).toBe(statusCodes.ok)
+
+        const $ = load(response.result)
+
+        expect($('html').attr('lang')).toBe(expectedLang)
```

Comment:
Testing the heading here will do the job of the two tests we're removing

```suggestion
        expect($('html').attr('lang')).toBe(expectedLang)
        expect($('h1').first().text()).toBe('Sites')
```